internal/js_printer/js_printer_test.go

```diff
@@ -932,13 +932,60 @@ func TestAvoidSlashScript(t *testing.T) {
+	expectPrinted(t, "x = (-Infinity).toString", "x = (-Infinity).toString;\n")
+	expectPrinted(t, "x = (Infinity) ** 2", "x = Infinity ** 2;\n")
+	expectPrinted(t, "x = (-Infinity) ** 2", "x = (-Infinity) ** 2;\n")
+	expectPrinted(t, "x = ~Infinity", "x = ~Infinity;\n")
+	expectPrinted(t, "x = ~-Infinity", "x = ~-Infinity;\n")
+	expectPrinted(t, "x = Infinity * y", "x = Infinity * y;\n")
+	expectPrinted(t, "x = Infinity / y", "x = Infinity / y;\n")
+	expectPrinted(t, "x = y * Infinity", "x = y * Infinity;\n")
+	expectPrinted(t, "x = y / Infinity", "x = y / Infinity;\n")
+	expectPrinted(t, "throw Infinity", "throw Infinity;\n")
+
+	expectPrintedMinify(t, "x = Infinity", "x=Infinity;")
+	expectPrintedMinify(t, "x = -Infinity", "x=-Infinity;")
+	expectPrintedMinify(t, "x = (Infinity).toString", "x=Infinity.toString;")
+	expectPrintedMinify(t, "x = (-Infinity).toString", "x=(-Infinity).toString;")
+	expectPrintedMinify(t, "x = (Infinity) ** 2", "x=Infinity**2;")
+	expectPrintedMinify(t, "x = (-Infinity) ** 2", "x=(-Infinity)**2;")
+	expectPrintedMinify(t, "x = ~Infinity", "x=~Infinity;")
+	expectPrintedMinify(t, "x = ~-Infinity", "x=~-Infinity;")
+	expectPrintedMinify(t, "x = Infinity * y", "x=Infinity*y;")
+	expectPrintedMinify(t, "x = Infinity / y", "x=Infinity/y;")
+	expectPrintedMinify(t, "x = y * Infinity", "x=y*Infinity;")
+	expectPrintedMinify(t, "x = y / Infinity", "x=y/Infinity;")
+	expectPrintedMinify(t, "throw Infinity", "throw Infinity;")
+
+	expectPrintedMangle(t, "x = Infinity", "x = 1 / 0;\n")
+	expectPrintedMangle(t, "x = -Infinity", "x = -1 / 0;\n")
+	expectPrintedMangle(t, "x = (Infinity).toString", "x = (1 / 0).toString;\n")
+	expectPrintedMangle(t, "x = (-Infinity).toString", "x = (-1 / 0).toString;\n")
+	expectPrintedMangle(t, "x = Infinity ** 2", "x = (1 / 0) ** 2;\n")
+	expectPrintedMangle(t, "x = (-Infinity) ** 2", "x = (-1 / 0) ** 2;\n")
+	expectPrintedMangle(t, "x = ~Infinity", "x = ~(1 / 0);\n")
+	expectPrintedMangle(t, "x = ~-Infinity", "x = ~(-1 / 0);\n")
+	expectPrintedMangle(t, "x = Infinity * y", "x = 1 / 0 * y;\n")
+	expectPrintedMangle(t, "x = Infinity / y", "x = 1 / 0 / y;\n")
+	expectPrintedMangle(t, "x = y * Infinity", "x = y * (1 / 0);\n")
+	expectPrintedMangle(t, "x = y / Infinity", "x = y / (1 / 0);\n")
+	expectPrintedMangle(t, "throw Infinity", "throw 1 / 0;\n")
+
 	expectPrintedMangleMinify(t, "x = Infinity", "x=1/0;")
-	expectPrintedMangleMinify(t, "x = -Infinity+2", "x=-1/0+2;")
-	expectPrintedMangleMinify(t, "x = (-Infinity+2) / 3", "x=(-1/0+2)/3;")
-	expectPrintedMangleMinify(t, "1 + -Infinity", "1+-1/0;")
-	expectPrintedMangleMinify(t, "1 - -Infinity", "1- -1/0;")
+	expectPrintedMangleMinify(t, "x = -Infinity", "x=-1/0;")
+	expectPrintedMangleMinify(t, "x = (Infinity).toString", "x=(1/0).toString;")
+	expectPrintedMangleMinify(t, "x = (-Infinity).toString", "x=(-1/0).toString;")
+	expectPrintedMangleMinify(t, "x = Infinity ** 2", "x=(1/0)**2;")
```

Comment:
I've might overlook this but `1/0 ** 2` will result in Infinity as well. Can the parentheses be "turned off" for different `level`'s? Or will that cause more problems 😅.

Reply:
The expressions `(a / b) ** c` and `a / (b ** c)` have different ASTs. Having minification turn one into the other is not generally correct, even if it's correct in that specific case. For example, changing `console.log(0 ** Infinity ** 1);` into `console.log(0 ** 1 / 0 ** 1);` changes the output from `0` to `NaN`.

Reply:
Got it, thanks for the explanation 🎉